### PR TITLE
Publish build tools into wazuh-packages 

### PR DESCRIPTION
**Pull request by DFolchA**
Hello team, 

This PR adds the following content to wazuh-packages:  

- [x] How to make a Wazuh WPK
- [x] How to make a Kibana app
- [x] How to make a Splunk app
- [x] How to make an OVA

I have also modified the already existing READMEs by adding a global readme for the repository with links to each of the READMEs of the other tools.

Best regards,
Daniel Folch

### CODE REVIEW

File: aix/README.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+        `# ./generate_package.sh -b v3.8.2 -s /tmp -a x86_64 -r 3821`.
+    
+3. When the execution finishes, you can find your `.ppc.rpm` package in specified folder.
+
+## More Packages
+
+- [RPM](/rpms/README.md)
+- [Debian](/debs/README.md)
+- [MacOS](/macos/README.md)
```
Comment (alberpilot) [inline aix/README.md]: Please change all "MacOS" references to "macOS"

File: README.md [hunk 2]
```diff
@@ -8,85 +8,18 @@ Wazuh
 
 Wazuh is an Open Source Host-based Intrusion Detection System that performs log analysis, file integrity monitoring, policy monitoring, rootkit detection, real-time alerting, active response, vulnerability detector, etc.
 
-In this repository, you can find the necessary tools to build a Wazuh package for Debian based OS (.deb) and an RPM based OS package (.rpm).
+In this repository, you can find the necessary tools to build a Wazuh package for Debian based OS (.deb), RPM based OS package (.rpm), MacOS (.macos) and AIX (.aix), the OVA, and the apps for Kibana and Splunk:
```
Comment (alberpilot) [inline README.md]: The file extensions are incorrect and necessaries.

File: splunkapp/README.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+## Tools needed to build the package
+
+To build a Wazuh Splunk app package you need to install the following tools:
+  - `docker`: [installation guide](https://docs.docker.com/install/linux/docker-ce/centos/) for RPM and [installation guide](https://docs.docker.com/install/linux/docker-ce/debian/) for Debian.
+- `git`:  [installation guide](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git). 
+  
+## Building Wazuh Splunk app packages
+
+To build a Splunk app package, you need to download this repository and use the `generate_wazuh_splunk_app.sh` script. This script will download the source code from the [wazuh/wazuh-splunk](https://github.com/wazuh/wazuh-splunk) repository build a Docker image with all the necessary tools to build the app package and run a Docker container from that image that will generate a `.tar.gz` package.
```
Comment (alberpilot) [inline splunkapp/README.md]: Please review if a "," is necessary between "will download the source code from the repository build a Docker. Review all places with this sentence.

File: ova/README.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      -v, --version          [Required] Version of wazuh to install on VM.
+      -e, --elastic-version  [Required] Elastic version to download inside VM.
+      -r, --repository       [Required] Status of the packages [stable/unstable]
+      -c, --clean            [Optional] Clean the local machine.
+      -h, --help             [  Util  ] Show this help.
+
+```
+
+  
+ To build an OVA with versiohn 3.9.0 using elastic 6.2.2 and the stable repositories you can use:
```
Comment (alberpilot) [inline ova/README.md]: Please fix 6.2.2 version

File: ova/README.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+In this repository, you can find the necessary tools to build your own OVA file with all Wazuh componets installed.
+
+## Tools needed to build the OVA:
+
+To build an OVA you need to install the following tools:
+- `Virtual Box`: [instalation guide](https://www.virtualbox.org/manual/UserManual.html#installation)
+- `Vagrant`: [instalation guide](https://www.vagrantup.com/docs/installation/)
+- `Git`:  [installation guide](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git). 
+- `Python`
```
Comment (alberpilot) [inline ova/README.md]: There is no installation guide for Python. Please add a guide.

File: macos/README.md [hunk 6]
```diff
@@ -13,30 +13,30 @@ In this repository, you can find the necessary tools to build a Wazuh package fo
 ## Tools needed to build the package
 
 To build a Wazuh package you need to install the following tools:
-  - [Packages](http://s.sudre.free.fr/Software/Packages/about.html): You can install this on OSX using the generate_wazuh_packages.sh script in this repo.
-  - `git`: on OSX install with homebrew use `brew install git`
+- [Packages](http://s.sudre.free.fr/Software/Packages/about.html): You can install this on OSX using the generate_wazuh_packages.sh script in this repo.
```
Comment (alberpilot) [inline macos/README.md]: Please change OSX to macOS. All references.

File: ova/README.md (hunk 5 above)
Comment (DFolchA) [reply ova/README.md]: Done in https://github.com/wazuh/wazuh-packages/pull/162/commits/99bd73a341721442e0fc065982bd83743368562d

File: macos/README.md (hunk 6 above)
Comment (DFolchA) [reply macos/README.md]: Done in https://github.com/wazuh/wazuh-packages/pull/162/commits/4c4b8cfd1a572c0993e15c0f7e953f2f249a79a0

File: ova/README.md (hunk 4 above)
Comment (DFolchA) [reply ova/README.md]: Done in https://github.com/wazuh/wazuh-packages/pull/162/commits/99bd73a341721442e0fc065982bd83743368562d

File: splunkapp/README.md (hunk 3 above)
Comment (DFolchA) [reply splunkapp/README.md]: Done in https://github.com/wazuh/wazuh-packages/pull/162/commits/24140971527228f1b10c43b13ac492b6e8409fda

File: aix/README.md (hunk 1 above)
Comment (DFolchA) [reply aix/README.md]: Done in https://github.com/wazuh/wazuh-packages/pull/162/commits/a03ea419f6431886c0f5b99fdb607af85436ba36

File: README.md (hunk 2 above)
Comment (DFolchA) [reply README.md]: Done in https://github.com/wazuh/wazuh-packages/pull/162/commits/a03ea419f6431886c0f5b99fdb607af85436ba36

File: README.md [hunk 7]
```diff
@@ -8,85 +8,18 @@ Wazuh
 
 Wazuh is an Open Source Host-based Intrusion Detection System that performs log analysis, file integrity monitoring, policy monitoring, rootkit detection, real-time alerting, active response, vulnerability detector, etc.
 
-In this repository, you can find the necessary tools to build a Wazuh package for Debian based OS (.deb) and an RPM based OS package (.rpm).
+In this repository, you can find the necessary tools to build a Wazuh package for Debian based OS, RPM based OS package, MacOS, AIX, the OVA, and the apps for Kibana and Splunk:
```
Comment (BraulioV) [inline README.md]: Change `MacOS` to `macOS`.

File: aix/README.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
+Wazuh
+=====
+
+[![Slack](https://img.shields.io/badge/slack-join-blue.svg)](https://wazuh.com/community/join-us-on-slack/)
+[![Email](https://img.shields.io/badge/email-join-blue.svg)](https://groups.google.com/forum/#!forum/wazuh)
+[![Documentation](https://img.shields.io/badge/docs-view-green.svg)](https://documentation.wazuh.com)
+[![Documentation](https://img.shields.io/badge/web-view-green.svg)](https://wazuh.com)
+
+Wazuh is an Open Source Host-based Intrusion Detection System that performs log analysis, file integrity monitoring, policy monitoring, rootkit detection, real-time alerting, active response, vulnerability detector, etc.
```
Comment (BraulioV) [inline aix/README.md]: This introduction is not really necessary. This has been already said in the main README.md file.

File: aix/README.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+1. Download this repository and go to the rpm directory:
+    ```bash
+    $ curl -L https://github.com/wazuh/wazuh-packages/tarball/master | tar zx 
+    $ cd wazuh-wazuh-packages-*
+    $ cd aix
+    ```
+
+2. Execute the `generate_rpm_package.sh` script to build the package. There are multiple parameters to select which package is going to be built, its architecture, etc. Here you can see all the different parameters:
+    ```shellsession
+    #./generate_package.sh -h
```
Comment (BraulioV) [inline aix/README.md]: A space is missing between `#`  and `./`.

File: aix/README.md [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+[![Documentation](https://img.shields.io/badge/docs-view-green.svg)](https://documentation.wazuh.com)
+[![Documentation](https://img.shields.io/badge/web-view-green.svg)](https://wazuh.com)
+
+Wazuh is an Open Source Host-based Intrusion Detection System that performs log analysis, file integrity monitoring, policy monitoring, rootkit detection, real-time alerting, active response, vulnerability detector, etc.
+
+In this repository, you can find the necessary tools to build a Wazuh package for AIX.
+
+## Building RPM packages
+
+To build an AIX package, you need to download this repository and use the `generate_wazuh_packages.sh` script. This script will download the source code from the [wazuh/wazuh](https://github.com/wazuh/wazuh) repository, and generate the `.ppc.rpm` package.
```
Comment (BraulioV) [inline aix/README.md]: I think it is a good idea to said that the script can install the build dependencies too.

File: debs/README.md [hunk 11]
```diff
@@ -0,0 +1,72 @@
+Wazuh
+=====
+
+[![Slack](https://img.shields.io/badge/slack-join-blue.svg)](https://wazuh.com/community/join-us-on-slack/)
+[![Email](https://img.shields.io/badge/email-join-blue.svg)](https://groups.google.com/forum/#!forum/wazuh)
+[![Documentation](https://img.shields.io/badge/docs-view-green.svg)](https://documentation.wazuh.com)
+[![Documentation](https://img.shields.io/badge/web-view-green.svg)](https://wazuh.com)
+
+Wazuh is an Open Source Host-based Intrusion Detection System that performs log analysis, file integrity monitoring, policy monitoring, rootkit detection, real-time alerting, active response, vulnerability detector, etc.
```
Comment (BraulioV) [inline debs/README.md]: Same as before, I don't think that this is necessary.

File: debs/README.md [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+Building a .deb package is pretty similar to build a .rpm package. You need to download the repository and execute the `generate_debian_package.sh` script. This will create the docker image, choose the needed files to build the package and build it.
+
+1. Download this repository and go to the debs directory:
+    ```bash
+    $ git clone https://github.com/wazuh/wazuh-packages && cd wazuh-packages/debs
+    ```
+
+2. Execute the `generate_debian_package.sh` script to built the package. There are multiple parameters to select which package is going to be build, its architecture, etc. Here you can see all the different parameters:
+      ```shellsession
+      #./generate_debian_package.sh -h
```
Comment (BraulioV) [inline debs/README.md]: A white space is missing.

File: debs/README.md [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+          -r, --revision <rev>      [Required] Package revision that append to version e.g. x.x.x-rev
+          -j, --jobs <number>       [Optional] Number of parallel jobs when compiling.
+          -p, --path <path>         [Optional] Installation path for the package. By default: /var/ossec.
+          -d, --debug               [Optional] Build the binaries with debug symbols. By default: no.
+          -h, --help                Show this help.
+
+      ```
+    * To build a wazuh-manager package for amd64 (x86_64) and store it in `/tmp`:
+        `# ./generate_debian_package.sh -b 3.9 -s /tmp -t manager -a amd64 -r 0`.
+    * To build a wazuh-agent package in `/home/ec2-user` for i386 with release 2:
```
Comment (BraulioV) [inline debs/README.md]: Change the user to a generic one.

File: macos/README.md [hunk 14]
```diff
@@ -8,35 +8,35 @@ Wazuh
...
-In this repository, you can find the necessary tools to build a Wazuh package for OSX.
+In this repository, you can find the necessary tools to build a Wazuh package for macOS.
 
 ## Tools needed to build the package
 
 To build a Wazuh package you need to install the following tools:
-  - [Packages](http://s.sudre.free.fr/Software/Packages/about.html): You can install this on OSX using the generate_wazuh_packages.sh script in this repo.
-  - `git`: on OSX install with homebrew use `brew install git`
+- [Packages](http://s.sudre.free.fr/Software/Packages/about.html): You can install this on macOS using the generate_wazuh_packages.sh script in this repo.
+- `git`: on macOS install with homebrew use `brew install git`
```
Comment (BraulioV) [inline macos/README.md]: If you need `brew` to install `git`, add the link to install `brew`.

File: macos/README.md [hunk 15]
```diff
@@ -8,35 +8,35 @@ Wazuh
...
 
-## Building OSX packages
+## Building macOS packages
 
-To build an OSX package, you need to download this repository and use the `generate_wazuh_packages.sh` script. This script will download the source code from the [wazuh/wazuh](https://github.com/wazuh/wazuh) repository and automatize the package generation.
+To build an macOS package, you need to download this repository and use the `generate_wazuh_packages.sh` script. This script will download the source code from the [wazuh/wazuh](https://github.com/wazuh/wazuh) repository and automatize the package generation.
```
Comment (BraulioV) [inline macos/README.md]: Change the last part from `...repository and automatize the package generation.` to `...repository and build the package`.

File: macos/README.md [hunk 16]
```diff
@@ -49,18 +49,27 @@ To build an OSX package, you need to download this repository and use the `gener
          
 **important detail:** the building of certain branches fails due to incorrect values on the `VERSION` file on the source which generate missing specs. To make a correct building it is recommended to use exacts tags instead of branches. For example, to build a package for Wazuh 3.7 you should use ``` -v v3.7.2 ```
```
Comment (BraulioV) [inline macos/README.md]: Remove this line.

File: macos/README.md [hunk 17]
```diff
@@ -49,18 +49,27 @@ To build an OSX package, you need to download this repository and use the `gener
...
 
-Use the `generate_wazuh_packages.sh` script for build packages for OSX.
+Use the `generate_wazuh_packages.sh` script for build packages for macOS.
 
 The `package_files` contains some files used by the Buildpackages tool to generate the package. The most important file is wazuh-agent-pkgproj, which is used by Buildpackages to generate the package and have to be updated by the script with the specs and default Wazuh configurations. Also, there are two scripts, `postinstall.sh` and `preinstall.sh` that are loaded in the package to be executed during the installation.
 
 The specs folder contains the pkgproj files which are used to generate the wazuh-agent.pkgproj file. Sometimes a spec file (pkgproj) for certain version is missing and then, the `generate_wazuh_packages.sh` script generates it based on the most recent version. If that doesn't work maybe it could be because the most recent version has changes that don't work for the compiled one. In this case, a possible solution would be to manually copy the immediately preceding version of the spec and change the version to the right one (Attention: you must change the version in different parts of the file).
```
Comment (BraulioV) [inline macos/README.md]: Remove this line too.

File: ova/Ova2Ovf.py [hunk 18]
```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python
+
+# Neova Health
```
Comment (BraulioV) [inline ova/Ova2Ovf.py]: I would change the header of this file. The original file was from Neova Health, but we have modified a bit the script. Give the credits to Neova Health and said that it has been modified by Wazuh.

File: ova/README.md [hunk 19]
```diff
@@ -0,0 +1,64 @@
...
+- `Vagrant`: [instalation guide](https://www.vagrantup.com/docs/installation/)
+- `Git`:  [installation guide](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git). 
+- `Python`: [installation guide](https://www.python.org/download/releases/2.7/)
+
+
+## Building OVA file:
+
+To generate the ova file, you need to download this repository and use the `generate_ova.sh` script. This script will create a vagrant box and provision it to be ready to use wazuh, as specified on provision.sh, and then export it to an Ova file.
+
+```shellsesion
```
Comment (BraulioV) [inline ova/README.md]: This shellsession is inconsistent with the next one. Add an indent to this or remove the indentation from the following code block.

File: ova/generate_ova.sh [hunk 20]
```diff
@@ -0,0 +1,189 @@
+#!/bin/bash
+# Program to upload packages to a development repository.
+# Copyright (c) 2016-2018 Wazuh, Inc <support@wazuh.com>
```
Comment (BraulioV) [inline ova/generate_ova.sh]: The copyright date is wrong. Also, change the email to `info@wazuh.com`.

File: rpms/README.md [hunk 21]
```diff
@@ -0,0 +1,71 @@
...
+To build an RPM package, you need to download this repository and use the `generate_wazuh_rpm.sh` script. This script will download the source code from the [wazuh/wazuh](https://github.com/wazuh/wazuh) repository or the [wazuh/wazuh-api](https://github.com/wazuh/wazuh-api) (depending on which package do you want to build), select automatically the RPM spec file, build a Docker image with all the necessary tools to build the RPM package and run a Docker container from that image that will generate the `.src.rpm` and `.rpm` packages.
+
+1. Download this repository and go to the rpm directory:
+    ```bash
+    $ git clone https://github.com/wazuh/wazuh-packages && cd wazuh-packages/rpms
+    ```
+
+2. Execute the `generate_rpm_package.sh` script to build the package. There are multiple parameters to select which package is going to be built, its architecture, etc. Here you can see all the different parameters:
+    ```shellsession
+    #./generate_rpm_package.sh -h
```
Comment (BraulioV) [inline rpms/README.md]: A white space is missing.

File: splunkapp/README.md [hunk 22]
```diff
@@ -0,0 +1,67 @@
...
+To build a Splunk app package, you need to download this repository and use the `generate_wazuh_splunk_app.sh` script. This script will download the source code from the [wazuh/wazuh-splunk](https://github.com/wazuh/wazuh-splunk) repository, build a Docker image with all the necessary tools to build the app package and run a Docker container from that image that will generate a `.tar.gz` package.
+
+1. Download this repository and go to the splunkapp directory:
+    ```bash
+    $ git clone https://github.com/wazuh/wazuh-packages && cd wazuh-packages/splunkapp
+    ```
+
+2. Execute the `generate_wazuh_splunk_app.sh` script to build the package. There are multiple parameters to select: which version of the app is going to be built, where is going to be stored in, etc. Here you can see all the different parameters:
+    ```shellsession
+    $./generate_wazuh_splunk_app.sh -h
```
Comment (BraulioV) [inline splunkapp/README.md]: A white space is missing.

File: wpk/wpkpack.py [hunk 23]
```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
+# June 26, 2017
```
Comment (BraulioV) [inline wpk/wpkpack.py]: Change the date

File: README.md (hunk 7 above)
Comment (alberpilot) [reply README.md]: Agree

File: aix/README.md (hunk 8 above)
Comment (alberpilot) [reply aix/README.md]: Agree

File: aix/README.md (hunk 9 above)
Comment (alberpilot) [reply aix/README.md]: +1

File: aix/README.md (hunk 10 above)
Comment (alberpilot) [reply aix/README.md]: Indeed, and the cleaning option

File: debs/README.md (hunk 11 above)
Comment (alberpilot) [reply debs/README.md]: +1

File: ova/Ova2Ovf.py (hunk 18 above)
Comment (alberpilot) [reply ova/Ova2Ovf.py]: Totally, I don't know when we changed the header but yes, give the credits to Neova Health and Wazuh modifications

**Pull request merged**